src/minmax.rs

```diff
@@ -0,0 +1,154 @@
+use crate::position::PositionSimd;
+use crate::SIMD_LEN;
+use std::simd::prelude::SimdPartialEq;
+use std::simd::Mask;
+use std::simd::Simd;
+use std::simd::SimdElement;
+use std::slice;
+
+pub trait MinMaxSimd<'a, T>
+where
+    T: PartialOrd + Ord + Copy,
```

Comment:
The `Ord` bound here prevents floats from being used (also in `MaxSimd` and `MinSimd`).

I'm not sure if it makes sense to mirror the traits with `Float` variants (for example, `MinMaxFloatSimd`), use declarative macros to implement functions, or something else.

Using declarative macros for the separate implementations like `impl_minmax_integers!(u8, i8, ...)` and `impl_minmax_floats!(f32, f64)` would allow for bringing in the appropriate trait for `simd::num::SimdFloat` in the latter impl while keeping the simplicity of the current implementation for integers.

Reply:
Don't worry floats are coming! I just started with the non floats. Not sure how I'll go about it quite yet. Personally I'm not a huge rust macro fan but will see!

Reply:
Thanks, and take your time.

> Personally I'm not a huge rust macro fan but will see!

I don't blame you, I'm not either 🙂

My own experience with `portable_simd` has been more concrete uses and not generic interfaces. This ended up being a more delicate design situation than I gave any thought to initially.